### PR TITLE
pkg/k8s: add conversion for DeleteFinalStateUnknown objects for CiliumNode

### PR DESCRIPTION
**Pull request by aanm**
:warning: :warning: ~**This PR is being merged against #8460 for easier review**~ :warning: :warning: 

~Do not merge until #8460 is merged first.~

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/8461)

### CODE REVIEW

File: pkg/k8s/factory_functions.go [hunk 1]
```diff
@@ -529,3 +529,13 @@ func ConvertToNamespace(obj interface{}) interface{} {
...
 	}
 }
+
+// ConvertToCiliumNode converts a *cilium_v2.CiliumNode into a
+// *cilium_v2.CiliumNode or a cache.DeletedFinalStateUnknown into
+// a cache.DeletedFinalStateUnknown with a *cilium_v2.CiliumNode in its Obj.
+// If the given obj can't be cast into either *cilium_v2.CiliumNode
+// nor cache.DeletedFinalStateUnknown, the original obj is returned.
+func ConvertToCiliumNode(obj interface{}) interface{} {
+	cnp, _ := obj.(*cilium_v2.CiliumNode)
```
Comment (ianvernon) [inline pkg/k8s/factory_functions.go]: Nit: rename returned value from cnp --> node

**Review by ianvernon — APPROVED** [review]
Small nit, nothing blocking.

File: pkg/k8s/factory_functions.go (hunk 1 above)
Comment (aanm) [reply pkg/k8s/factory_functions.go]: It's simply a code move. It was like this as before.

**Comment by aanm** [comment]
test-me-please

**Review by tgraf — APPROVED** [review]

**Pull request merged**